src/librustc_mir/util/pretty.rs

```diff
@@ -596,8 +596,8 @@ pub fn write_allocations<'tcx>(
         write!(w, "\nalloc{}", id)?;
         let alloc = tcx.alloc_map.lock().get(id);
         match alloc {
-            None => writeln!(w, " dangling")?,
-            Some(GlobalAlloc::Function(inst)) => writeln!(w, " (fn: {})", inst)?,
+            None => write!(w, " dangling")?,
```

Comment:
I think for const-eval this is unreachable, but for Miri's printing we probably still need this? But the code here only checks the global map, so Miri cannot share this anyway?

Reply:
Miri doesn't share this at all, it only shares the byte/reloc printing logic.

This is probably unreachable, but I felt that we should rather end up having something printed than not, even in the presence of broken state.

Reply:
Okay. But then I'd print "deallocated" or so, Pointers to nonexisting allocations are dangling, but I never heard of "dangling allocations".